complete some things, simple type names, misc

supersaiyansubtlety opened the pull request:
- add several simple_type_field_names
- remove `RegistryKey` from simple_type_field_names (they're often named after their generic type)
- map all `RegistryKey`s affected by its removal from simple_type_field_names (no lost coverage! [except in unmapped classes])
- complete n.m.village
- complete n.m.registry
- complete `Items` (thanks type names for 1000+ settings)
- rename/fix some things in `IdListUtil`
- make users of `IdListUtil::createByIdGetter` consistent
- miscellaneous mappings mapped along the way

supersaiyansubtlety committed:
complete SharedConstants

supersaiyansubtlety committed:
complete n.m.village

supersaiyansubtlety committed:
remove 'registryKey' from simple_type_field_names

supersaiyansubtlety committed:
registry work

supersaiyansubtlety committed:
complete n.m.registry.tag, more registry work

supersaiyansubtlety committed:
explicitly map CommandTreeUpdateS2CPacket consturctor param to work around proposer conflicts, make simple_type_field_names_path an input of insertAutoGeneratedMappings task, make 'id' simple type name exclusive

supersaiyansubtlety committed:
complete n.m.registry

OroArmor commented on simple_type_field_names.json5:
```diff
@@ -27,10 +29,20 @@
   // Registries
-  "net/minecraft/unmapped/C_tqxyjqsk": "registry",
-  "net/minecraft/unmapped/C_xhhleach": "registryKey",
+  "net/minecraft/unmapped/C_tqxyjqsk": {
```
```suggestion
  "net/minecraft/unmapped/C_tqxyjqsk": { // Registry
```

OroArmor commented on simple_type_field_names.json5:
```diff
@@ -27,10 +29,20 @@
   "net/minecraft/unmapped/C_wqxmvzdq": "registryManager", // DynamicRegistryManager
   "net/minecraft/unmapped/C_hqoyyfco": "context", // BootstrapContext
+  "net/minecraft/unmapped/C_ednuhnnn": "tag", // TagKey
+  "net/minecraft/unmapped/C_ncpywfca": {
```
```suggestion
  "net/minecraft/unmapped/C_ncpywfca": { // Identifier
```

OroArmor commented on simple_type_field_names.json5:
```diff
@@ -98,6 +110,15 @@
   },
 
+  // Village
+  "net/minecraft/unmapped/C_argwryfu": {
```
```suggestion
  "net/minecraft/unmapped/C_argwryfu": { // TradeOffer
```

OroArmor commented on simple_type_field_names.json5:
```diff
@@ -138,6 +159,12 @@
 
   // Network
   "net/minecraft/unmapped/C_idfydwco": "buf", // PacketByteBuf
+  "net/minecraft/unmapped/C_mnjlaveb$C_mlxyrlwl": {
```
```suggestion
  "net/minecraft/unmapped/C_mnjlaveb$C_mlxyrlwl": { // TagNetworkSerialization$NetworkPayload
```
~~How many places is this a parameter or field?~~
Nice catch, this is actually in quite a few places.

supersaiyansubtlety committed:
Add class comments to new simple type names

Co-authored-by: Eli Orona <eliorona@live.com>

supersaiyansubtlety committed:
map ~97/270 RegsitryKeys affected by their removal from simple type names
rename IdListUtil#sortArray -> createByIdGetter
make enums that use createByIdGetter consistent
restore SpawnReason
add simple type names for Entity+LivinEntity
add stub FindInvalidSimpleTypeFieldNamesTask class
map a few other things along the way

supersaiyansubtlety committed:
complete Items with the help of simple type names

supersaiyansubtlety committed:
half way there! (through RegistryKeys affected by their removal from simple type names)

supersaiyansubtlety committed:
finished! fixing up RegistryKeys affected by their removal from simple type names

supersaiyansubtlety committed:
fix typos

supersaiyansubtlety committed:
remove FindInvalidSimpleTypeFieldNamesTask stub for separate PR

supersaiyansubtlety commented on mappings/net/minecraft/server/function/FunctionGetter.mapping:
```diff
@@ -0,0 +1,6 @@
+CLASS net/minecraft/unmapped/C_uxzryfxv net/minecraft/server/function/FunctionGetter
+	FIELD f_cxzaybvc commandFunctionSet Z
+	FIELD f_kkregdwp commandFunction Ljava/util/Optional;
+	METHOD m_ashjquub getid ()Lnet/minecraft/unmapped/C_ncpywfca;
```
spell check fail?

supersaiyansubtlety committed:
fixup a few things folling self-review

OroArmor commented on build.gradle:
```diff
@@ -116,9 +117,16 @@ clean.doFirst {
 tasks.build.dependsOn "compressTiny", "tinyJar", "v2UnmergedMappingsJar", "v2MergedMappingsJar"
 
 task insertAutoGeneratedMappings(dependsOn: [buildMappingsTiny, downloadPerVersionMappings], type: AddProposedMappingsTask) {
+	final enigmaProfile = file("enigma_profile.json")
+
+	inputs.file(
+		new JsonSlurper().parse(enigmaProfile)
+			.services.jar_indexer.args.simple_type_field_names_path
+	)
+
```
This... Works for now. I would make an issue once this is merged to do this properly.

supersaiyansubtlety replied:
~~afaict a proper fix would require enigma plugins to have a way of listing files they depend on, then enigma would list all files all plugins depend on~~
<sub>hadn't found `getServiceProfiles`</sub>

Is that what you mean?
Or something in qmap buildSrc?

OroArmor replied:
something in the buildsrc

supersaiyansubtlety committed:
make simple_type_field_names an input of insertAutoGeneratedMappings betterly

eristhea commented on mappings/net/minecraft/block/TrialSpawnerConfig.mapping:
```diff
@@ -1,9 +1,41 @@
+		FIELD f_jgmutzza ticksBetweenSpawn I
+		FIELD f_jqtdptyp simultaneousMobsAddedPerPlayer F
+		FIELD f_nblcrtjb totalMobsAddedPerPlayer F
+		FIELD f_reubvwnb itemsToDropWhenOminous Lnet/minecraft/unmapped/C_xhhleach;
```
since we're not exactly matching the loot table name here, i think we could do better -- something like `ominousRewards` or `ominousItemRewards`

supersaiyansubtlety replied:
It comes from the `TrialSpawnerConfig` codec:
```java
RegistryKey.createCodec(RegistryKeys.LOOT_TABLE).optionalFieldOf("items_to_drop_when_ominous", DEFAULT_INSTANCE.itemsToDropWhenOminous).forGetter(TrialSpawnerConfig::itemsToDropWhenOminous)
```
The `TrialSpawnerConfig` field+method are proposed, I just made the builder match them.

I agree it's not the greatest name though.

eristhea commented on mappings/net/minecraft/command/argument/ResourceKeyArgument.mapping:
```diff
@@ -3,20 +3,16 @@ CLASS net/minecraft/unmapped/C_zgptntbw net/minecraft/command/argument/ResourceK
 	FIELD f_fmnxgqhj INVALID_FEATURE_EXCEPTION Lcom/mojang/brigadier/exceptions/DynamicCommandExceptionType;
 	FIELD f_hqteeflf INVALID_JIGSAW_EXCEPTION Lcom/mojang/brigadier/exceptions/DynamicCommandExceptionType;
 	FIELD f_nufqqyoc EXAMPLES Ljava/util/Collection;
-	FIELD f_umcxcanc key Lnet/minecraft/unmapped/C_xhhleach;
```
why are these being removed?

eristhea commented on simple_type_field_names.json5:
```diff
@@ -27,10 +29,20 @@
 
   // Registries
-  "net/minecraft/unmapped/C_tqxyjqsk": "registry",
-  "net/minecraft/unmapped/C_xhhleach": "registryKey",
```
this appears to be removed with a replacement
![image](https://github.com/user-attachments/assets/f44b6337-8e04-4ccc-8250-4053fd2ca1bb)

supersaiyansubtlety replied:
oops guess I missed that one

eristhea commented on mappings/net/minecraft/util/ErrorReporter.mapping:
```diff
@@ -3,12 +3,12 @@ CLASS net/minecraft/unmapped/C_jtpvewkp net/minecraft/util/ErrorReporter
+		METHOD m_fdjwfdmg copyErrorsByPath ()Lcom/google/common/collect/Multimap;
 		METHOD m_oscjmxcu getPath ()Ljava/lang/String;
-		METHOD m_pznzwrnj getErrors ()Ljava/util/Optional;
+		METHOD m_pznzwrnj getFormatedErrors ()Ljava/util/Optional;
```
```suggestion
		METHOD m_pznzwrnj getFormattedErrors ()Ljava/util/Optional;
```

eristhea reviewed (changes requested):
looks fantastic, just a couple things:
- there are quite a few empty mappings here, i think `dropInvalidMappings` should address that
- though the enum refactors are great, I don't agree with the `BY_ID` -> `GET_BY_ID` refactor. i don't see any good reason to add the extra word.

supersaiyansubtlety commented:
> though the enum refactors are great, I don't agree with the BY_ID -> GET_BY_ID refactor. i don't see any good reason to add the extra word.

Yeah I'm a bit torn on `byId` and `GET_BY_ID`:
- strictly by convention they should both have "get"
- but I prefer the brevity without "get" and I don't think clarity is lost
- I usually like to name fields holding functional interface objects extra function-y because they're slightly counter-intuitive to me, hence the "get" on the field
- on the other hand if "by id" is function-y enough for the method it should be function-y enough for the field 
- they also should definitely match

I guess "by id" is fine for both.

supersaiyansubtlety committed:
fix typo

Co-authored-by: ix0rai <ix0rai64@gmail.com>

supersaiyansubtlety committed:
drop invalid mappings

supersaiyansubtlety committed:
re-complete ResourceKeyArgument

supersaiyansubtlety committed:
rename all GET_BY_ID -> BY_ID

supersaiyansubtlety committed:
Merge branch '24w39a' into procrastination
